src/ProxyClient/Http/HttpAdapterInterface.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace FOS\HttpCache\ProxyClient\Http;
+
+use FOS\HttpCache\Exception\ExceptionCollection;
+
+/**
+ * An adapter to work with Httplug.
+ *
+ * @author David Buchmann <mail@davidbu.ch>
+ */
+interface HttpAdapterInterface
```

Comment:
not sure if we should call this adapter, but turns out we do need some layer between the proxy client and Httplug, or we would need to duplicate logic or share it in a base class.